packages/frontend/src/hooks/scheduler/useSchedulersDeleteMutation.ts

```diff
@@ -0,0 +1,32 @@
+import { ApiError } from '@lightdash/common';
+import { useMutation, useQueryClient } from 'react-query';
+import { lightdashApi } from '../../api';
+import useToaster from '../toaster/useToaster';
+
+const deleteScheduler = async (uuid: string) =>
+    lightdashApi<undefined>({
+        url: `/schedulers/${uuid}`,
+        method: 'DELETE',
+        body: undefined,
+    });
+
+export const useSchedulersDeleteMutation = () => {
+    const queryClient = useQueryClient();
+    const { showToastSuccess, showToastError } = useToaster();
+    return useMutation<undefined, ApiError, string>(deleteScheduler, {
+        mutationKey: ['delete_scheduler'],
+        onSuccess: async () => {
+            await queryClient.invalidateQueries('chart_schedulers');
+            await queryClient.invalidateQueries('dashboard_schedulers');
+            showToastSuccess({
+                title: `Success! Scheduled delivery was delete.`,
+            });
+        },
+        onError: (error) => {
+            showToastError({
+                title: `Failed to delete scheduled delivery`,
+                subtitle: error.error.message,
+            });
+        },
+    });
+};
```

Comment:
I really like separating hooks in the same files and grouping by folder.
One thing I was thinking is that actually extracting toasts and UI related things from the queries and mutations.

what can be a better pattern is to have `useSchedulersDeleteMutation` (and other hooks) as well to proxy `useMutation` `useQuery` option. and showing toasts and UI related things can be done in the component side. example of passing mutation options  `useCreateMutation` can be found in a `useExplore` hook for example.

I just feel like UI related stuff should not be inside mutations, including redirects and showing toasts. I am open to discussion if you think otherwise.